internal/mq/msgstream/mqwrapper/kafka/kafka_consumer.go

```diff
@@ -12,34 +12,68 @@ import (
 )
 
 type Consumer struct {
-	c               *kafka.Consumer
-	config          *kafka.ConfigMap
-	msgChannel      chan mqwrapper.Message
-	hasSeek         bool
-	hasConsume      bool
-	skipMsg         bool
-	latestMsgOffset kafka.Offset
-	topic           string
-	groupID         string
-	closeCh         chan struct{}
-	chanOnce        sync.Once
-	closeOnce       sync.Once
+	c          *kafka.Consumer
+	config     *kafka.ConfigMap
+	msgChannel chan mqwrapper.Message
+	hasAssign  bool
+	skipMsg    bool
+	topic      string
+	groupID    string
+	chanOnce   sync.Once
+	closeOnce  sync.Once
+	closeCh    chan struct{}
 }
 
-func newKafkaConsumer(config *kafka.ConfigMap, topic string, groupID string) (*Consumer, error) {
-	closeCh := make(chan struct{})
-	msgChannel := make(chan mqwrapper.Message, 256)
+const timeout = 3000
 
-	kafkaConsumer := &Consumer{
+func newKafkaConsumer(config *kafka.ConfigMap, topic string, groupID string, position mqwrapper.SubscriptionInitialPosition) (*Consumer, error) {
+	msgChannel := make(chan mqwrapper.Message, 256)
+	kc := &Consumer{
 		config:     config,
 		msgChannel: msgChannel,
 		topic:      topic,
 		groupID:    groupID,
-		closeCh:    closeCh,
+		closeCh:    make(chan struct{}),
+	}
+
+	err := kc.createKafkaConsumer()
+	if err != nil {
+		return nil, err
+	}
+
+	// if it's unknown, we leave the assign to seek
+	if position != mqwrapper.SubscriptionPositionUnknown {
+		var offset kafka.Offset
+		if position == mqwrapper.SubscriptionPositionEarliest {
+			offset, err = kafka.NewOffset("earliest")
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			latestMsgID, err := kc.GetLatestMsgID()
+			if err != nil {
+				return nil, err
+			}
+			offset = kafka.Offset(latestMsgID.(*kafkaID).messageID)
+			kc.skipMsg = true
+		}
+
+		start := time.Now()
+		topicPartition := []kafka.TopicPartition{{Topic: &topic, Partition: mqwrapper.DefaultPartitionIdx, Offset: offset}}
+		err = kc.c.Assign(topicPartition)
```

Comment:
It is possible the topic or partition does not exist when the consumer of Datacoord starts firstly.